gedit_wakatime/wakatime.py

```diff
@@ -6,33 +6,37 @@
 
 logger = logging.getLogger('gedit-wakatime-plugin')
 
-ACTION_FREQUENCY = 2
+ACTION_FREQUENCY = 2 * 60
 PLUGIN_USER_AGENT = 'gedit/3 gedit-wakatime/{}'.format(__version__)
 
-_last_heartbeat = {}
+_last_heartbeat_time = 0
+_last_heartbeat_file = None
 
-def enough_time_passed(now, last_time):
-    if now - last_time > ACTION_FREQUENCY * 60:
-        return True
-    return False
 
 def flatten_args(args_dict):
     return [str(arg) for key, value in args_dict.items() if value
                 for arg in [key, value] if not isinstance(arg, bool)]
 
 def send_heartbeat(file_uri, write=False):
-    global _last_heartbeat
-    logger.debug('Sending heartbeat')
+    global _last_heartbeat_time, _last_heartbeat_file
+    logger.debug("Heartbeat triggered")
     now = time.time()
-    if enough_time_passed(now, _last_heartbeat.get(file_uri, 0)) or write:
-        waka_args = {
-            '--file': file_uri,
-            '--time': now,
-            '--write': write,
-            '--plugin': PLUGIN_USER_AGENT,
-        }
-        waka_args = ['wakatime'] + flatten_args(waka_args)
-        logger.debug('Calling wakatime with: {}'.format(waka_args))
-        subprocess.Popen(waka_args)
-        _last_heartbeat[file_uri] = now
-
+    if (
+        not write
+        and file_uri == _last_heartbeat_file
+        and now - _last_heartbeat_time < ACTION_FREQUENCY
+    ):
+        logger.debug("Not necessary to send heartbeat")
```

Comment:
This will generate too many logs.

Reply:
This is only debug logs, so it's printed only with `env GEDIT_WAKATIME_PLUGIN_DEBUG=1`, and there are two other messages from this plugin emitted in the same sequence anyway. Should this really be commented out?

Reply:
I think it's dangerous to add anything that runs on every cursor movement. It could easily start slowing down the UI.

Reply:
Looks like the plugin already has debug logs on every cursor change, so this one more is fine too.